timelapse-colorizer-data/data_writer_utils.py

```diff
@@ -319,6 +323,26 @@ def sanitize_key_name(name: str) -> str:
     return re.sub(pattern, "", name)
 
 
+def generate_frame_paths(
+    num_frames: int,
+    start_frame: int = 0,
+    file_prefix: str = DEFAULT_FRAME_PREFIX,
+    file_suffix: str = DEFAULT_FRAME_SUFFIX,
+) -> List[str]:
+    """
+    Returns a list of image file paths of length `num_frames`, with a configurable prefix, suffix,
+    and starting frame number.
+
+    By default, returns the list:
+    - `frame_0.png`
+    - `frame_1.png`
+    - `frame_2.png`
+    - `...`
+    - `frame_{num_frames - 1}.png`
+    """
+    return [file_prefix + str(i + start_frame) + file_suffix for i in range(num_frames)]
```

Comment:
Here's why I was confused in the meeting earlier when this came up.  Why do we even care what the frame paths start with? They are fully specified in the data and so it doesn't matter one bit whether the file names are 0 based or start_frame based.   No one should ever even be looking at these filenames.

Reply:
The issue comes in from `write_image`, which takes in a frame number and writes the file `frame_{num}.png`. The problem was that datasets would directly pass in the frame numbers/timepoint from the dataset (say, frame #9) and then the image would get written as `frame_9.png`. (There's no way to track what frame numbers have been written by the writer directly because of multiprocessing)

If frame #9 is the first frame in the dataset, the behavior before (which wrote `frame_0.png, frame_1.png, ...`) would expect frames 0-8 to exist as images in the dataset. This caused datasets to fail to load, as expected frames would be missing.

Reply:
Aha! thanks. Makes sense.

Reply:
I wonder if write_frame could automatically insert into the array of strings the same way it inserts bounding boxes.

Reply:
We could, but the user would need to pass in the string array to `make_frame`, like we're doing for bounding boxes. This would also potentially introduce some more complexity... That array of frame paths would only be present if we did the `make_frames` step, which means we'd need a fallback if the user is running the scripts with `--noframes`.

OH. Unrelated but I just realized there will be a bunch of datasets missing bounding box information because of the `--noframes` option. I'll write up an issue ticket about this and hopefully explain it a bit better.

Reply:
Aforementioned issue ticket: https://github.com/allen-cell-animated/colorizer-data/issues/22

Reply:
The thing that I think is funky and am worried that it might have to change again, is that it is easy for the image processing to write out names that don't match with the results of generate_paths.

Reply:
We should think about making --no-frames just simply leave the list of frame paths unmodified.  The first time a dataset is prepared you need to generate the frames anyway.

Reply:
Agreed that this is going to be a potential issue, and I've update #22 with more detail.